### PR TITLE
Use new clusterResourceOverridesEnabled flag

### PR DESCRIPTION
**Pull request by spadgett**
If `clusterResourceOverridesEnabled` in the web console config is set,
assume that CPU request, CPU limit, and memory request are all
calculated. Don't try to validate these fields. Also remove request to
limit conversions for HPAs. A default DEFAULT_HPA_CPU_TARGET_PERCENT can
still be set by an extension if needed.

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres Not ready for review, but I wanted to get a PR open as I work on this.

@benjaminapetersen This likely conflicts with the HPA changes you're making.

**Comment by spadgett** [comment]
/assign @jwforres 

PTAL. I added some unit tests.

File: app/scripts/constants.js [hunk 1]
```diff
@@ -666,5 +670,17 @@ angular.extend(window.OPENSHIFT_CONSTANTS, {
...
+    'openshift',
+    'kubernetes',
+    'kube'
+  ],
+
+  CLUSTER_RESOURCE_OVERRIDES_EXEMPT_PROJECT_PREFIXES: [
+    'openshift-',
+    'kubernetes-',
+    'kube-'
+  ]
```
[the remark this answers is not in the document]
Comment (spadgett) [reply app/scripts/constants.js]: Yes, it uses the list here:

https://github.com/spadgett/origin/blob/5b71e8d8e441a0d39f9228a4bce4c7a6a13be08b/pkg/project/registry/projectrequest/delegated/delegated.go#L82-L85

Admission controller code:

https://github.com/spadgett/origin/blob/5b71e8d8e441a0d39f9228a4bce4c7a6a13be08b/pkg/quota/admission/clusterresourceoverride/admission.go#L153-L165

File: app/scripts/controllers/create/createFromImage.js [hunk 2]
```diff
@@ -225,9 +226,7 @@ angular.module("openshiftConsole")
         }
 
         var validatePodLimits = function() {
-          if (!$scope.hideCPU) {
```
[the remark this answers is not in the document]
Comment (spadgett) [reply app/scripts/controllers/create/createFromImage.js]: I don't think it strictly was. It was meant to be an optimization, skipping a check we didn't need to make.

**Comment by spadgett** [comment]
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1474196

**Comment by jwforres** [comment]
generally lgtm, let me know when its ready for a final review

**Comment by spadgett** [comment]
@jwforres I added some additional tests for validation with and without cluster resource overrides.

File: test/spec/services/limitsRangesServiceSpec.js [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+          }
+        }
+      }];
+
+      var cpuProblems = LimitRangesService.validatePodLimits(mockLimitRanges, 'cpu', containers, mockProject);
+      expect(cpuProblems.length).toBeGreaterThan(0);
+    });
+
+    it("should report problems when cpu limit is too low", function() {
+      var containers = [{
```
Comment (jwforres) [inline test/spec/services/limitsRangesServiceSpec.js]: you redefine these same sets of containers objects a bunch of times, maybe define them above as `lowCPULimit` and `highCPULimit` close to where the limit range is defined ?  And then its easier to validate the tests are right?

**Comment by spadgett** [comment]
updated

**Pull request merged**